src/global_distrib/mod_global_distrib_outgoing_conns_sup.erl

```diff
@@ -33,21 +33,27 @@ add_server(Server) ->
 
 -spec get_connection(Server :: ejabberd:lserver()) -> pid().
 get_connection(Server) ->
-    case whereis(mod_global_distrib_utils:server_to_sup_name(Server)) of
-        undefined -> ok = add_server(Server);
-        _ -> ok
-    end,
+    ensure_server_started(Server),
     mod_global_distrib_server_sup:get_connection(Server).
```

Comment:
hmm, there can be more than one connection between servers and you pick a random one?
No wonder this test fails

http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1680/4000/pgsql_mnesia.20.0/big/ct_run.test@travis-job-c416160e-42d3-4434-ade5-ca16966418d8.2018-01-22_16.46.45/ejabberd_tests.tests.mod_global_distrib_SUITE.logs/run.2018-01-22_17.02.33/mod_global_distrib_suite.test_in_order_messages_on_multiple_connections_with_bounce.html

Why are you sure that the stanzas would be delivered in order?

Reply:
`get_connection` chain returns a random connection; the connection is then bound to the client process that initiated the call: https://github.com/esl/MongooseIM/blob/master/src/global_distrib/mod_global_distrib.erl#L137

Reply:
It should actually match on already_started as well (in add_server). This Is going to be changed.